bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyThingCreator.java

```diff
@@ -606,7 +609,8 @@ private static String getThingTypeID(String serviceName, String deviceType, Stri
         if (serviceNameLowerCase.startsWith(THING_TYPE_SHELLY2_PREFIX)) { // Shelly v2
             return getRelayOrRollerType(THING_TYPE_SHELLY2_RELAY_STR, THING_TYPE_SHELLY2_ROLLER_STR, mode);
         }
-        if (serviceNameLowerCase.startsWith(THING_TYPE_SHELLYPLUG_STR) && !serviceNameLowerCase.contains("plugus")) {
+        if (serviceNameLowerCase.startsWith(THING_TYPE_SHELLYPLUG_STR) && !serviceNameLowerCase.contains("plugus")
+                && !serviceNameLowerCase.contains("plugsg3")) {
```

Comment:
is this nessesary?

Reply:
Without this the thing type would be `shellyplugs` and not `shellyplusplug` causing it to use Gen1 instead of Gen2 APIs.

Reply:
you right :-)